src/main/java/edu/ucsb/cs156/dining/controllers/ReviewsController.java

```diff
@@ -0,0 +1,100 @@
+import edu.ucsb.cs156.dining.entities.UCSBDiningCommons;
+import edu.ucsb.cs156.dining.errors.EntityNotFoundException;
+import edu.ucsb.cs156.dining.repositories.ReviewsRepository;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import org.springframework.security.web.bind.annotation.AuthenticationPrincipal;
+
+import java.util.ArrayList;
+import jakarta.validation.Valid;
+
+/**
+ * This is a REST controller for Reviews
+ */
+
+@Tag(name = "Reviews")
+@RequestMapping("/api/reviews")
+@RestController
+@Slf4j
+public class ReviewsController extends ApiController {
+
+    @Autowired
+    ReviewsRepository reviewsRepository;
+
+    /**
+     * THis method returns a list of all reviews.
+     * @return a list of all reviews
+     */
+    @Operation(summary= "List all ucsb reviews of dining commons")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<Reviews> allReviews() {
+        Iterable<Reviews> reviews = reviewsRepository.findAll();
+        return reviews;
+    }
+
+    /**
+     * This method creates a new review. Accessible only to users with the role "ROLE_ADMIN".
+     * @param student_id studentID of the reviewer
+     * @param item_id itemID of the review
+     * @param date_served date served 
+     * @param status status of the review
+     * @param moderator_comments comments from the moderator
+     * @param created_date created date of the review
+     * @param last_edited_date last edited date of the review
+     * @return the save review
+     */
+    @Operation(summary= "Create a new review")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping("/post")
+    public Reviews postReview(
+        @Parameter(name="student_id") @RequestParam int student_id,
+        @Parameter(name="item_id") @RequestParam int item_id,
+        @Parameter(name="date_served") @RequestParam String date_served,
+        @Parameter(name="status") @RequestParam(required=false) String status,
+        @Parameter(name="moderator_comments") @RequestParam(required=false) String moderator_comments,
+        @Parameter(name="created_date") @RequestParam String created_date,
```

Comment:
This should be assigned by code in the endpoint, and should NOT be something that the user can modify.  Calculate this from the current time INSIDE the controller action, or better yet, use the automatic mechanism for putting this on the record as demonstrated here:

https://github.com/ucsb-cs156/proj-happycows/blob/7a5456e61fffafa73a3464eb16360c4aa06cd50f/src/main/java/edu/ucsb/cs156/happiercows/entities/ChatMessage.java#L29C3-L32C28

```
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date timestamp;
```

Note that to get this to work, you also need to add this line in the main class file as shown here:

https://github.com/ucsb-cs156/proj-happycows/blob/7a5456e61fffafa73a3464eb16360c4aa06cd50f/src/main/java/edu/ucsb/cs156/happiercows/HappierCowsApplication.java#L23C1-L23C64

```
@EnableJpaAuditing(dateTimeProviderRef = "utcDateTimeProvider")   
```

And this method in the main class:


https://github.com/ucsb-cs156/proj-happycows/blob/7a5456e61fffafa73a3464eb16360c4aa06cd50f/src/main/java/edu/ucsb/cs156/happiercows/HappierCowsApplication.java#L60

```java
 @Bean
  public DateTimeProvider utcDateTimeProvider() {
      return () -> {
        ZonedDateTime now = ZonedDateTime.now();
        return Optional.of(now);
      };
  }
```